iceoryx_posh/include/iceoryx_posh/internal/roudi/service_registry.hpp

```diff
@@ -32,23 +35,44 @@ static const capro::IdString_t Wildcard{"*"};
 class ServiceRegistry
 {
   public:
-    static constexpr uint32_t MAX_INSTANCES_PER_SERVICE = 100u;
-    using InstanceSet_t = cxx::vector<capro::IdString_t, MAX_INSTANCES_PER_SERVICE>;
-    struct instance_t
+    enum class Error
     {
-        InstanceSet_t instanceSet;
+        INVALID_STATE,
+        SERVICE_REGISTRY_FULL,
     };
-    using serviceMap_t = std::map<capro::IdString_t, instance_t>;
+    /// @todo #415 should be connected with iox::MAX_NUMBER_OF_SERVICES
+    static constexpr uint32_t MAX_SERVICE_DESCRIPTIONS = 100U;
+    using ReferenceCounter_t = uint64_t;
+    using ServiceDescriptionVector_t =
+        cxx::vector<cxx::pair<capro::ServiceDescription, ReferenceCounter_t>, MAX_SERVICE_DESCRIPTIONS>;
```

Comment:
The `cxx::pair` produces hard to read code. While I was reviewing I constantly had to look up what `.first` and `.second` meant. Instead of using a `cxx::pair` could you create a small struct like:
```cpp
struct ServiceDescriptionEntry {
  capro::ServiceDescription serviceDescription{};
  ReferenceCounter_t referenceCounter = 0U;
};
```

Then the code would be much more readable.

Reply:
Completely agree! Done.